### PR TITLE
Relax libhdf version and update CI to test system libhdf5

### PR DESCRIPTION
**Pull request by musm**
Summary of changes:

- Relax libhdf5 to v1.10.4 (this is the version shipped with ubuntu-lts)
- Merge all CI into one file (easier to maintain and update)
- Test libhdf5 + mpio integration on CI

### CODE REVIEW

**Comment by musm** [comment]
@jipolanco  I'm running into 
```
Attempting to use an MPI routine before initializing MPICH
HDF5: infinite loop closing library
      L,T_top,P,P,Z,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD
```
when I run the test with with MPI.  This is  on my system with:
```
sudo apt install libhdf5-mpich-dev
export JULIA_HDF5_LIBRARY_PATH=/usr/lib/x86_64-linux-gnu/hdf5/mpich/
```

Any chance you would know why we are running into this?

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -3,7 +3,30 @@ name: CI
...
+
+  test1:
+    name: Julia ${{ matrix.version }} - ${{ matrix.os }} - ${{ matrix.arch }} (system libhdf5)
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        version:
+          - '1.3'
+        os:
+          - ubuntu-20.04
```
Comment (musm) [inline .github/workflows/CI.yml:15]: change to `ubuntu-latest` when gh-actions get's updated (required for libhdf5 v1.10.4)

**Comment by jipolanco** [comment]
> @jipolanco I'm running into
> 
> ```
> Attempting to use an MPI routine before initializing MPICH
> HDF5: infinite loop closing library
>       L,T_top,P,P,Z,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD,FD
> ```
> 
> when I run the test with with MPI. This is on my system with:
> 
> ```
> sudo apt install libhdf5-mpich-dev
> export JULIA_HDF5_LIBRARY_PATH=/usr/lib/x86_64-linux-gnu/hdf5/mpich/
> ```
> 
> Any chance you would know why we are running into this?
> 
> In particular it looks like this method is triggering the error
> 
> ```
>     HDF5.h5p_set_fapl_mpio(fileprop, comm, info)
> ```
> 
> in the `mpio.jl` test suite.

@musm I'm not sure, but it may be because you're not using your system's MPICH. For this, you should set the environment variable `JULIA_MPI_BINARY=system` and then rebuild MPI.jl and HDF5.jl.

You may also be interested in this [github CI setup](https://github.com/jipolanco/PencilArrays.jl/blob/master/.github/workflows/ci.yml) using MPI and HDF5 on Ubuntu.

**Comment by musm** [comment]
That was very helpful, many thanks @jipolanco

**Comment by jipolanco** [comment]
No problem. Thanks for the extra tests and for relaxing the libhdf5 requirement!

**Pull request merged**